src/main/scala/mrtjp/projectred/expansion/TileProjectBench.scala

```diff
@@ -234,21 +264,32 @@ class SlotProjectCrafting(
 
     val invCrafting = new InventoryCrafting(new NodeContainer, 3, 3)
     for (i <- 0 until 9)
-      invCrafting.setInventorySlotContents(i, tile.currentInputs(i))
+      invCrafting.setInventorySlotContents(i, tile.tInputs(i))
     FMLCommonHandler
       .instance()
       .firePlayerCraftingEvent(player, stack, invCrafting)
 
     tile.updateRecipe()
   }
 
-  def searchFor(
+  private var start = 0
+  private var end = 0
+
+  private def searchFor(
       world: World,
       recipe: IRecipe,
       inputs: Array[ItemStack],
       storage: Array[ItemStack]
   ): Boolean = {
-    i = 0
+
+    if (tile.isPlanRecipe) { // search in resources inventory
+      start = 0
+      end = 18
+    } else { // search in crafting grid inventory
+      start = 18
```

Comment:
According to https://github.com/GTNewHorizons/ProjectRed/blob/6040cd81fc5d58b7dba27929acc658e83089676c/src/main/scala/mrtjp/projectred/expansion/TileProjectBench.scala#L86 the crafting grid is from slot 0-8, is that not the case here?

Reply:
I can be sure it's not a problem, but I have plan to rewrite the crafting logic, because it's confused

Reply:
I need time to explain all of this things xd